ui/src/components/date/QasDate.vue

```diff
@@ -140,38 +149,88 @@ export default {
       }
     },
 
-    normalizedEvents () {
-      return this.useUnmaskEvents ? this.getUnmaskedList(this.events) : this.events
-    },
-
     normalizedOptions () {
       return this.useUnmaskOptions ? this.getUnmaskedList(this.options) : this.options
     }
   },
 
   watch: {
     events () {
-      if (this.useInactiveDates) this.setInactiveEvents(this.currentDate)
+      this.setEvents(this.currentDate)
     }
   },
 
   mounted () {
+    this.dateElement = this.$refs.date.$el
+
+    // muda estilo da navegação
     this.setNewNavigatorDisplay()
 
-    if (this.useInactiveDates) {
-      this.currentDate = this.getCurrentDate()
-      this.setInactiveEvents(this.currentDate)
-    }
+    // observa as mudanças no DOM do QDate > .q-date__content
+    this.setObserveDate()
+
+    // seta data atual de acordo com a view do mês atual
+    this.setCurrentDate()
+
+    // seta os eventos, ativos e inativos
+    this.setEvents(this.currentDate)
+  },
+
+  unmounted () {
+    this.dateObserver.disconnect()
   },
 
   methods: {
+    createActiveEventElement (dayElement, eventElement) {
+      const buttonElement = dayElement.querySelector('.q-btn__content')
+
+      buttonElement?.appendChild?.(eventElement)
+    },
+
+    createCounterInactiveEventElement (dayElement, eventElement) {
+      const spanElement = document.createElement('span')
+      const [child] = dayElement.children
+
+      spanElement.setAttribute('data-event', true)
+      dayElement.setAttribute('data-has-inactive-event', true)
+
+      spanElement.classList.add('flex', 'items-center', 'justify-center', 'text-center')
+
+      child.setAttribute('data-day-content', child.textContent)
+
+      spanElement.appendChild(child)
+      spanElement.appendChild(eventElement)
+      dayElement.appendChild(spanElement)
+    },
+
+    createPointerEventElement (isInactive, currentColor, eventElement) {
+      const eventPointerElement = document.createElement('div')
+
+      eventPointerElement.classList.add(
+        'qas-date__event-pointer',
+        `bg-${isInactive ? 'grey-4' : currentColor || 'primary'}`
+      )
+
+      eventElement.appendChild(eventPointerElement)
+    },
+
+    getCurrentColor (date, color) {
+      return (
+        (this.isEventColorCallback ? this.eventColor(date) : this.eventColor) || color
+      )
+    },
```

Comment:
poderia ser na mesma linha `return (this.isEventColorCallback ? this.eventColor(date) : this.eventColor) || color`